tortoise/tortoise.go

```diff
@@ -725,17 +760,17 @@ func (t *turtle) onBallot(ballot *types.BallotTortoiseData) error {
 	return nil
 }
 
-func (t *turtle) compareBeacons(logger log.Log, bid types.BallotID, lid types.LayerID, beacon types.Beacon) (bool, error) {
+func (t *turtle) compareBeacons(bid types.BallotID, lid types.LayerID, beacon types.Beacon) (bool, error) {
 	epoch := t.epoch(lid.GetEpoch())
 	if epoch.beacon == nil {
 		return false, errBeaconUnavailable
 	}
 	if beacon != *epoch.beacon {
-		logger.With().Debug("ballot has different beacon",
-			log.Uint32("layer_id", lid.Uint32()),
-			log.Stringer("block", bid),
-			log.ShortStringer("ballot_beacon", beacon),
-			log.ShortStringer("epoch_beacon", epoch.beacon),
+		t.logger.Debug("ballot has different beacon",
+			zap.Uint32("layer_id", lid.Uint32()),
+			zap.Stringer("block", bid),
+			log.ZShortStringer("ballot_beacon", beacon),
+			log.ZShortStringer("epoch_beacon", epoch.beacon),
```

Comment:
Imo we don't need the `ZShortStringer` wrapper:

```suggestion
			zap.String("ballot_beacon", beacon.ShortString()),
			zap.String("epoch_beacon", epoch.beacon.ShortString()),
```

Reply:
we do need it

Reply:
> we do need it

Would you mind to explain why?

Reply:
beacon and some other things are stored as bytes , not string. in order to transform them into string, you need to encode them in hex. therefore such allocation should be performed lazily, only when needed

Reply:
Makes sense